From 0bfb3291ab3a678ef0de43a320170e7c6cecfa4b Mon Sep 17 00:00:00 2001
From: p_q <p_q@nebulae.co>
Date: Sat, 27 Sep 2025 11:15:10 +0200
Subject: [PATCH] tools/lemon.c: fix gcc15/C23 build.

In gcc15 the default standard used is C23.
In C23 declaring foo() means foo(void). Previously it acted as a wildcard for arg list.
This commit fixes function declarations with empty arg list in lemon.
Relevant doc: https://gcc.gnu.org/gcc-15/porting_to.html
---
 tools/lemon.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/tools/lemon.c b/tools/lemon.c
index b0f628d..497629a 100644
--- a/tools/lemon.c
+++ b/tools/lemon.c
@@ -59,12 +59,12 @@ static struct action *Action_new(void);
 static struct action *Action_sort(struct action *);
 
 /********** From the file "build.h" ************************************/
-void FindRulePrecedences();
-void FindFirstSets();
-void FindStates();
-void FindLinks();
-void FindFollowSets();
-void FindActions();
+void FindRulePrecedences(struct lemon *lemp);
+void FindFirstSets(struct lemon *lemp);
+void FindStates(struct lemon *lemp);
+void FindLinks(struct lemon *lemp);
+void FindFollowSets(struct lemon *lemp);
+void FindActions(struct lemon *lemp);
 
 /********* From the file "configlist.h" *********************************/
 void Configlist_init(void);
